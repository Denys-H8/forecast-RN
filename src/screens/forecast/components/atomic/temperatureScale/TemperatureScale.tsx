import React from 'react';
import { View } from 'react-native';
import { useSelector } from 'react-redux';

import CustomText from 'src/components/text';
import { styles } from './temperatureScale.styles';
import { temperatureScale } from 'src/store/settings/selectors';

type TProps = {
  maxTemperature: number;
  minTemperature: number;
};

const TemperatureScale: React.FC<TProps> = ({ maxTemperature, minTemperature }) => {
  const dif = maxTemperature - minTemperature;
  const arr = new Array(dif);
  const temperatureType = useSelector(temperatureScale);
  const rowWidth = temperatureType === 'metric' ? 6 : 3;

  return (
    <View style={styles.container}>
      <View style={styles.degreesContainer}>
        <CustomText style={styles.number}>{minTemperature}</CustomText>
        <CustomText style={styles.celsius}>0</CustomText>
      </View>
      {maxTemperature === minTemperature || dif <= 3 ? (
        <View style={[styles.scale, { width: 25 }]} />
      ) : (
        <View style={[styles.scale, { width: rowWidth * arr.length }]} />
      )}
      <View style={styles.degreesContainer}>
        <CustomText style={styles.number}>{maxTemperature}</CustomText>
        <CustomText style={styles.celsius}>0</CustomText>
      </View>
    </View>
  );
};

export default TemperatureScale;
