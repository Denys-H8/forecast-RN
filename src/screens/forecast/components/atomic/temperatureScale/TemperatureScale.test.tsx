import React from 'react';
import { render } from '@testing-library/react-native';
import TemperatureScale from './TemperatureScale';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureStore([]);

const store = mockStore({
  settingsReducer: {
    temperatureScale: 'metric',
  },
});

const storeImperial = mockStore({
  settingsReducer: {
    temperatureScale: 'imperial',
  },
});

describe('TemperatureScale ', () => {
  test('should render metric and temp diff > 3', () => {
    const maxTemperature = 27;
    const minTemperature = 15;
    const { toJSON } = render(
      <Provider store={store}>
        <TemperatureScale minTemperature={minTemperature} maxTemperature={maxTemperature} />
      </Provider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render imperial and temp diff <= 3', () => {
    const maxTemperature = 16;
    const minTemperature = 14;
    const { toJSON } = render(
      <Provider store={storeImperial}>
        <TemperatureScale minTemperature={minTemperature} maxTemperature={maxTemperature} />
      </Provider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
