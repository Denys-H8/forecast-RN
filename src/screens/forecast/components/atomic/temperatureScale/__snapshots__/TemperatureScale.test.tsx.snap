// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemperatureScale  should render imperial and temp diff <= 3 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "WorkSans-Regular",
          },
          Object {
            "color": "#fff",
            "fontSize": 32,
            "lineHeight": 36,
          },
        ]
      }
    >
      14
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "WorkSans-Regular",
          },
          Object {
            "color": "#fff",
            "fontSize": 18,
            "lineHeight": 18,
          },
        ]
      }
    >
      0
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#8B8EA5",
          "borderRadius": 15,
          "height": 12,
          "marginHorizontal": 5,
          "maxWidth": 85,
        },
        Object {
          "width": 25,
        },
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "WorkSans-Regular",
          },
          Object {
            "color": "#fff",
            "fontSize": 32,
            "lineHeight": 36,
          },
        ]
      }
    >
      16
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "WorkSans-Regular",
          },
          Object {
            "color": "#fff",
            "fontSize": 18,
            "lineHeight": 18,
          },
        ]
      }
    >
      0
    </Text>
  </View>
</View>
`;

exports[`TemperatureScale  should render metric and temp diff > 3 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "WorkSans-Regular",
          },
          Object {
            "color": "#fff",
            "fontSize": 32,
            "lineHeight": 36,
          },
        ]
      }
    >
      15
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "WorkSans-Regular",
          },
          Object {
            "color": "#fff",
            "fontSize": 18,
            "lineHeight": 18,
          },
        ]
      }
    >
      0
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#8B8EA5",
          "borderRadius": 15,
          "height": 12,
          "marginHorizontal": 5,
          "maxWidth": 85,
        },
        Object {
          "width": 72,
        },
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "WorkSans-Regular",
          },
          Object {
            "color": "#fff",
            "fontSize": 32,
            "lineHeight": 36,
          },
        ]
      }
    >
      27
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "WorkSans-Regular",
          },
          Object {
            "color": "#fff",
            "fontSize": 18,
            "lineHeight": 18,
          },
        ]
      }
    >
      0
    </Text>
  </View>
</View>
`;
